### PR TITLE
#163512197 Users should be able to edit and delete their comments 

### PR DESCRIPTION
**Pull request by davishooly**
**What does this PR do?**
-  Implements user ability to delete comments on articles.
- Implements user ability to update comments on articles.

**Description of tasks to be completed?**
- Authenticated users can delete and update the comments they have posted.

**How should this be manually tested?**

- Clone the repo and cd into the project folder
- Switch to the appropriate branch with `ft-delete-update-comments-163512197`
- Run `npm install` to update your packages.
- Run tests with `npm test`
- Specify the domain you are running the app from with `export REACT_APP_API="https://ah-codetitans-staging.herokuapp.com/"`
- Fire up the server with `npm start`
- Login to the app.
- Navigate to `/article/best-python-version` and post a new comment on the article.
- Once the single article loads, click on the share buttons to share.

**What are the relevant pivotal tracker stories?**
- [#163512197](https://www.pivotaltracker.com/story/show/163512197)

**Checklist (edited)**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

**Screenshots**
screen shot 2019-01-30 at 19 54 36

### CODE REVIEW

**Review by bettblake08 — APPROVED** [review]
👍

**Review by denomash — APPROVED** [review]
👍

**Review by jillo-abdullahi — APPROVED** [review]
👍

**Review by Bryan-Cee — APPROVED** [review]
👍 LGTM

**Review by Darius-Ndubi — APPROVED** [review]
👍

File: src/components/containers/comment.js [hunk 1]
```diff
@@ -39,9 +41,31 @@ class Comment extends Component {
     return true;
   };
 
+  // update comment section
+  editComment = (event) => {
+    const { comment, actions } = this.props;
+    const { replyComment } = this.state;
+    const { id } = comment;
+
+    if (event.which === 13 || event.keyCode === 13) {
```
Comment (justmesam) [inline src/components/containers/comment.js]: `event.which` and `event.keyCode` are deprecated and might fail to work in different browsers so just find an alternative for this.

I recommend [KeyboardEvent.code](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code)

**Review by justmesam — CHANGES_REQUESTED** [review]

File: src/components/containers/comment.js (hunk 1 above)
Comment (davishooly) [reply src/components/containers/comment.js]: Thanks, I'm looking into it

**Review by justmesam — APPROVED** [review]

**Pull request merged**